### PR TITLE
feat(data): add docs for working with multiple files with Storage and GraphQL API; updates to existing code examples

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Description of changes:
feat(data): add docs for working with multiple files with Storage and GraphQL API; updates to existing code examples

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 1]
```diff
@@ -442,50 +442,894 @@ const deletedSong = await API.graphql({
...
+await API.graphql({
+  query: mutations.deletePhotoAlbum,
+  variables: { input: photoAlbumDetails },
+});
+```
+
+</Block>
+</BlockSwitcher>
+
+## Complete Examples:
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:992]:
```suggestion
## Complete examples
```

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 2]
```diff
@@ -442,50 +442,894 @@ const deletedSong = await API.graphql({
 </Block>
 </BlockSwitcher>
 
-## Complete example
+## Working with multiple files:
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:445]:
```suggestion
## Working with multiple files
```

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 3]
```diff
@@ -442,50 +442,894 @@ const deletedSong = await API.graphql({
...
+
+You may want to add multiple files to a single record, such as a user profile with multiple images. 
+To do this, you can add a list of file keys to the record. The following example adds a list of file keys to a record:
+
+### Schema
```
Comment (renebrandel) [inline src/fragments/lib/graphqlapi/js/working-with-files.mdx:450]:
```suggestion
### GraphQL schema to associate a data model with multiple files
```

**Review by renebrandel — DISMISSED** [review]

**Review by svidgen — DISMISSED** [review]
Approving assuming the full sample app has been successfully run.

It would make for a good revision or followup to add a section and/or a quick line on each interaction explaining 1) why we recommend the patterns we do, and 2) what "gotchas" developers still need to watch out for. E.g.,

> Our recommended patterns minimize the occurrence of clients seeing records that point to non-existent files. This optimizes for read latency by ensuring clients *rarely* attempt to fetch a non-existent file from Storage. However, any app that deletes files can inherently cause records on-device to point to non-existent files.
>
> For example, "Device A" might get `Record ABC`, which points to `File_XYZ`. "Device B" might delete `File_XYZ` and update `Record ABC` accordingly. However, "Device A" is still using the old `Record ABC`. Even though the shared global state is correctly in sync at every stage, the individual device ("Device A") has a record that points to a non-existent file. Similar issues can conceivably occur for updates. (Depending on your app, some of these mismatches can be minimized *even more* with [realtime data / graphql subscriptions](https://pr-5539.d39agkzf0cwtdz.amplifyapp.com/lib/graphqlapi/subscribe-data/q/platform/js/)).
> 
> **So, it's important to understand when this can occur in your app and codify meaningful behavior around these edges.**

Something like that, but probably worded much more betterrer. 😅

File: src/fragments/lib/graphqlapi/js/working-with-files.mdx [hunk 4]
```diff
@@ -456,244 +1808,451 @@ import { GraphQLQuery } from "@aws-amplify/api";
...
+  async function getImagesForPhotoAlbum() {
+    try {
+      // Query the record to get the file keys:
+      const response = await API.graphql<GraphQLQuery<GetPhotoAlbumQuery>>({
+        query: queries.getPhotoAlbum,
+        variables: { id: currentPhotoAlbum.id },
+      });
+      const photoAlbum = response.data?.getPhotoAlbum;
+
+      // Ensure that the record has associated images:
```
[the remark this answers is not in the document]
Comment (david-mcafee) [reply src/fragments/lib/graphqlapi/js/working-with-files.mdx:2114]: Will use your other suggestion throughout

**Review by svidgen — APPROVED** [review]

**Comment by david-mcafee** [comment]
> Approving assuming the full sample app has been successfully run.
> 
> It would make for a good revision or followup to add a section and/or a quick line on each interaction explaining 1) why we recommend the patterns we do, and 2) what "gotchas" developers still need to watch out for. E.g.,
> 
> > Our recommended patterns minimize the occurrence of clients seeing records that point to non-existent files. This optimizes for read latency by ensuring clients _rarely_ attempt to fetch a non-existent file from Storage. However, any app that deletes files can inherently cause records on-device to point to non-existent files.
> > For example, "Device A" might get `Record ABC`, which points to `File_XYZ`. "Device B" might delete `File_XYZ` and update `Record ABC` accordingly. However, "Device A" is still using the old `Record ABC`. Even though the shared global state is correctly in sync at every stage, the individual device ("Device A") has a record that points to a non-existent file. Similar issues can conceivably occur for updates. (Depending on your app, some of these mismatches can be minimized _even more_ with [realtime data / graphql subscriptions](https://pr-5539.d39agkzf0cwtdz.amplifyapp.com/lib/graphqlapi/subscribe-data/q/platform/js/)).
> > **So, it's important to understand when this can occur in your app and codify meaningful behavior around these edges.**
> 
> Something like that, but probably worded much more betterrer. 😅

I'll add this with my next PR! :)

**Review by renebrandel — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**